sumo_ql/environment/communication_device.py

```diff
@@ -53,26 +55,51 @@ def update_stored_rewards(self, link: str, reward: int) -> None:
             self.__data[link].pop(0)
         self.__data[link].append(reward)
 
-    def get_expected_reward(self, link: str) -> float:
+    def get_expected_reward(self, link: str) -> np.ndarray:
         """This method returns the expected reward for the given link. If there's no reward stored for the given link,
-        the method returns 0. It returns an average of the stored rewards otherwise.
+        the method returns 0. It returns a list with the averages of the stored rewards otherwise.
            If the link is not part of the commDev's neighboring links, the method raises a RunTimeError.
 
         Args:
             link (str): String with the link ID to get the expected reward
 
         Returns:
-            float: Average of the stored rewards for the given link if there is data stored or 0.0 otherwise
+            list: List of averages of the stored rewards for the given link if there is data stored or 0.0 otherwise
         """
+        nobj = len(self.__environment.objectives.known_objectives)
+        
         if link not in self.__data.keys():
             raise RuntimeError("Link is not connected to commDev's node")
 
         if len(self.__data[link]) > 0:
-            return np.array(self.__data[link]).mean()
-
-        return 0.0
-
-    def get_outgoing_links_expected_rewards(self) -> Dict[str, float]:
+            data_mean = np.mean(self.__data[link], axis=0)
+            if len(data_mean) == nobj: # if the amount of data is correct
+                return data_mean
+            else:   
+                print(f"Size data mean doesn't match number of objectives for {link}")
+        else:
+            print(f"Empty link data list for {link}")
+
+        return np.zeros(shape=nobj)
+    
+    def get_graph_neighbours_interval(self, graph_neighbours_link: dict, current_step: int) -> list:
+        number_of_intervals = len(graph_neighbours_link)
+        i = 0
+        for interval in graph_neighbours_link:
+            if i == number_of_intervals-1:
+                if interval[0] <= current_step <= interval[1]:
+                    return graph_neighbours_link[interval]
+            else:
+                if interval[0] < current_step <= interval[1]:
+                    return graph_neighbours_link[interval]
+            i += 1
+        #print("Interval not found, returning empty list")
+        return []
+
+
+    # modificar para que, para cada link de saída, programa busca vizinhos no grafo no tempo atual da simulação e envia os dados
```

Comment:
Aqui seria um "to do" ou já foi feito?
```suggestion
    #TODO: modificar para que, para cada link de saída, programa busca vizinhos no grafo no tempo atual da simulação e envia os dados
```

Reply:
Já foi feito, eu tinha anotado o que deveria fazer para lembrar depois, mas acabei esquecendo de remover essa anotação

Reply:
Criei um commit para remover esse comentário